src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -225,7 +234,7 @@ public class RSessionProvider : IRSessionProvider {
             var oldBroker = _brokerProxy.Set(brokerClient);
             try {
                 BrokerChanging?.Invoke(this, EventArgs.Empty);
-                await SwitchBrokerAsync(oldBroker, cancellationToken, lockToken.Reentrancy);
+                await SwitchBrokerAsync(cts.Token, lockToken.Reentrancy);
                 oldBroker.Dispose();
 
                 if (brokerClient.IsRemote) {
```

Comment:
If (unlikely, but anyway) writing to console throws, we'll be setting back disposed oldBroker